packages/react-dom/src/server/ReactPartialRendererHooks.js

```diff
@@ -229,11 +229,14 @@ export function useReducer<S, A>(
   }
 }
 
+// a variable to store current component while we compare memo values
+let cachedCurrentlyRenderingComponent = null;
```

Comment:
Why is this top level rather than inside `useMemo`?

Reply:
it's one less var to be gced, reuses the location across runs. you mentioned the local var has a cost I assumed you'd want to avoid?

Reply:
maybe I should just trust javascript engines

Reply:
Declaring variables pointing to shared objects has no effect on GC — I assume you mean the stack? I said it's a hypothetical concern but I'm not actually sure it matters and I think local function variable is fine.

Reply:
I wish we had some way to validate these JS engine assumptions, it’d be interesting to see how often they’re true 🤔

Reply:
Well, having two checks is likely going to be more expensive than having one check, no matter what you do.